ts/Stock/Indicators/AO/AOIndicator.ts

```diff
@@ -0,0 +1,291 @@
+            AO: Array<Array<number>> = [], // 0- date, 1- Awesome Oscillator
+            xData: Array<number> = [],
+            yData: Array<number> = [],
+            high = 1,
+            low = 2,
+            shortSum = 0,
+            longSum = 0,
+            shortSMA: number, // Shorter Period SMA
+            longSMA: number, // Longer Period SMA
+            awesome: number,
+            shortLastIndex: number,
+            longLastIndex: number,
+            price: number,
+            i: number,
+            j: number;
+
+        if (
+            xVal.length <= longPeriod ||
+            !isArray(yVal[0]) ||
+            yVal[0].length !== 4
+        ) {
+            return;
+        }
+
+        for (i = 0; i < longPeriod - 1; i++) {
+            price = ((yVal[i] as any)[high] + (yVal[i] as any)[low]) / 2;
+            if (i >= longPeriod - shortPeriod) {
+                shortSum = correctFloat(shortSum + price);
+            }
+            longSum = correctFloat(longSum + price);
+        }
+
+        for (j = longPeriod - 1; j < yValLen; j++) {
+            price = ((yVal[j] as any)[high] + (yVal[j] as any)[low]) / 2;
+            shortSum = correctFloat(shortSum + price);
+            longSum = correctFloat(longSum + price);
+
+            shortSMA = shortSum / shortPeriod;
+            longSMA = longSum / longPeriod;
+
+            awesome = correctFloat(shortSMA - longSMA);
+
+            AO.push([xVal[j], awesome]);
+            xData.push(xVal[j]);
+            yData.push(awesome);
+
+            shortLastIndex = j + 1 - shortPeriod;
+            longLastIndex = j + 1 - longPeriod;
+
+            shortSum = correctFloat(
+                shortSum -
+                (
+                    (yVal[shortLastIndex] as any)[high] +
+                    (yVal[shortLastIndex] as any)[low]
+                ) / 2
+            );
+            longSum = correctFloat(
+                longSum -
+                (
+                    (yVal[longLastIndex] as any)[high] +
+                    (yVal[longLastIndex] as any)[low]
+                ) / 2
+            );
+        }
+
+
+        return {
+            values: AO,
+            xData: xData,
+            yData: yData
+        } as IndicatorValuesObject<TLinkedSeries>;
+    }
+}
+
+interface AOIndicator {
+    nameBase: string;
+    nameComponents: Array<string>;
+    pointClass: typeof AOPoint;
+    crispCol: ColumnSeries['crispCol'];
+    drawPoints: ColumnSeries['drawPoints'];
+    getColumnMetrics: ColumnSeries['getColumnMetrics'];
+    translate: ColumnSeries['translate'];
+}
+
+extend(AOIndicator.prototype, {
+    nameBase: 'AO',
+    nameComponents: (false as any),
+
+    // Columns support:
+    markerAttribs: (noop as any),
+    getColumnMetrics: H.seriesTypes.column.prototype.getColumnMetrics,
+    crispCol: H.seriesTypes.column.prototype.crispCol,
+    translate: H.seriesTypes.column.prototype.translate,
+    drawPoints: H.seriesTypes.column.prototype.drawPoints
```

Comment:
Instead of using the namespace with `H.seriesTypes.column.prototype` here, you should replace it with the series type `ColumnSeries.prototype`.